src/ssort/_bindings.py

```diff
@@ -1,802 +1,76 @@
-    """
-    yield from get_bindings(node.value)
-    yield from get_bindings(node.slice)
-
-
-@get_bindings.register(ast.Starred)
-def _get_bindings_for_starred(node):
-    """
-    ..code:: python
-
-        Starred(expr value, expr_context ctx)
-    """
-    yield from get_bindings(node.value)
-
-
-@get_bindings.register(ast.Name)
-def _get_bindings_for_name(node):
-
-    """
-    ..code:: python
-
-        Name(identifier id, expr_context ctx)
-    """
-    return
-    yield
-
-
-@get_bindings.register(ast.List)
-def _get_bindings_for_list(node):
-
-    """
-    ..code:: python
-
-        List(expr* elts, expr_context ctx)
-    """
-    for elt in node.elts:
-        yield from get_bindings(elt)
-
-
-@get_bindings.register(ast.Tuple)
-def _get_bindings_for_tuple(node):
-    """
-    ..code:: python
-
-        Tuple(expr* elts, expr_context ctx)
-
-    """
-    for elt in node.elts:
-        yield from get_bindings(elt)
+import ast
+from typing import Iterable
+
+from ssort._visitor import NodeVisitor, register_visitor
+
+
+class _BindingsNodeVisitor(NodeVisitor[str]):
+    @register_visitor(ast.FunctionDef, ast.AsyncFunctionDef)
+    def visit_function_def(
+        self, node: ast.FunctionDef | ast.AsyncFunctionDef
+    ) -> Iterable[str]:
+        for decorator in node.decorator_list:
+            yield from self.visit(decorator)
+        yield node.name
+        yield from self.visit(node.args)
+        if node.returns is not None:
+            yield from self.visit(node.returns)
+
+    @register_visitor(ast.ClassDef)
+    def visit_class_def(self, node: ast.ClassDef) -> Iterable[str]:
+        for decorator in node.decorator_list:
+            yield from self.visit(decorator)
+        for base in node.bases:
+            yield from self.visit(base)
+        for keyword in node.keywords:
+            yield from self.visit(keyword.value)
+        yield node.name
+
+    @register_visitor(ast.Import)
+    def visit_import(self, node: ast.Import) -> Iterable[str]:
+        for name in node.names:
+            if name.asname:
+                yield name.asname
+            else:
+                root, *rest = name.name.split(".", 1)
+                yield root
+
+    @register_visitor(ast.ImportFrom)
+    def visit_import_from(self, node: ast.ImportFrom) -> Iterable[str]:
+        for name in node.names:
+            yield name.asname if name.asname else name.name
+
+    @register_visitor(ast.Global)
+    def visit_global(self, node: ast.Global) -> Iterable[str]:
+        yield from node.names
+
+    @register_visitor(ast.Nonlocal)
+    def visit_nonlocal(self, node: ast.Nonlocal) -> Iterable[str]:
+        yield from node.names
+
+    @register_visitor(ast.Lambda)
+    def visit_lambda(self, node: ast.Lambda) -> Iterable[str]:
+        yield from self.visit(node.args)
+
+    @register_visitor(ast.Name)
+    def visit_name(self, node: ast.Name) -> Iterable[str]:
+        if isinstance(node.ctx, ast.Store):
+            yield node.id
```

Comment:
I like that this just handles everything I was attempting to do with `_flatten_target`.  Branching later is good.